### PR TITLE
Version update 0.4.0

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Updates the version

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,23 @@ All notable changes to this project will be documented in this file.
...
+
+### Removed
+
+### Fixed
+
+### Security
+
+### Dependencies
+
+## [0.4.0] - 2023-11-xx
```
Comment (NickGeneva) [inline CHANGELOG.md:25]: put the 13th here

**Review by NickGeneva — CHANGES_REQUESTED** [review]

**Comment by NickGeneva** [comment]
Please also update the docker versions in the readme to 23.11 before we forget.

**Comment by ktangsali** [comment]
> Please also update the docker versions in the readme to 23.11 before we forget.

Thanks! Done

**Review by NickGeneva — APPROVED** [review]

**Review by akshaysubr — APPROVED** [review]

**Pull request merged**